productcat/src/main/java/com/visenze/productcat/android/data/GetGAIDTask.java

```diff
@@ -0,0 +1,60 @@
+package com.visenze.productcat.android.data;
+
+import android.content.Context;
+import android.os.AsyncTask;
+
+import com.google.android.gms.ads.identifier.AdvertisingIdClient;
+import com.google.android.gms.common.GooglePlayServicesNotAvailableException;
+import com.google.android.gms.common.GooglePlayServicesRepairableException;
+
+import java.io.IOException;
+
+public class GetGAIDTask extends AsyncTask<Void, Void, String> {
+
+
+    private final OnTaskSuccess mListener;
+    private Context mContext;
+
+    public GetGAIDTask(Context context, OnTaskSuccess listener) {
+        mContext = context;
+        mListener = listener;
+    }
+
+    @Override
+    protected String doInBackground(Void... params) {
+        AdvertisingIdClient.Info idInfo = null;
+
+        try {
+            idInfo = AdvertisingIdClient.getAdvertisingIdInfo(mContext.getApplicationContext());
+        } catch (GooglePlayServicesNotAvailableException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        } catch (GooglePlayServicesRepairableException e) {
+            e.printStackTrace();
+        }
+
+        String gaid = null;
+
+        try {
+            gaid = idInfo.getId();
+        } catch (NullPointerException e) {
```

Comment:
eh why need to catch NullPointer, should put this gaid = XXX line right after idInfo = AdvertisingIdClient.getAdvertisingIdInfo line. Alternately check for null instead of catching exception like this

Reply:
changed already. idInfo may get null if there is no google service available on the phone.